mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoScaler.java

```diff
@@ -357,15 +358,19 @@ public void scaleUpStage(final int numCurrentWorkers, final int desiredWorkers,
 
       public int getDesiredWorkersForScaleDown(final int decrement, final int numCurrentWorkers) {
         final int desiredWorkers;
-        if (!stageSchedulingInfo.getScalingPolicy().isEnabled()) {
-          logger.warn("Job " + jobId + " stage " + stage + " is not scalable, can't decrement #workers by " + decrement);
+        final StageScalingPolicy scalingPolicy = stageSchedulingInfo.getScalingPolicy();
+        if (!scalingPolicy.isEnabled()) {
+          logger.warn("Job {} stage {} is not scalable, can't decrement #workers by {}", jobId, stage, decrement);
           return numCurrentWorkers;
         }
         if (numCurrentWorkers < 0 || decrement < 1) {
           logger.error("current number of workers({}) not known or decrement({}) < 1, will not scale down", numCurrentWorkers, decrement);
           return numCurrentWorkers;
+        } else if (!scalingPolicy.isAllowScaleDownDuringEvacuated() && failoverStatusClient.getFailoverStatus() == FailoverStatusClient.FailoverStatus.REGION_EVACUEE) {
```

Comment:
I think it would be better to separate out the AutoscalingPolicy similar to sources and sinks in mantis and have a Netflix-specific implementation in nfmantis. These changes seem to be specific to Netflix. However, it's not clear how much of a change that is and whether its feasible in the short term.

Reply:
I think it's a good idea to refactor autoscaling policies for easy management.
That said, I'd argue that this isn't specific to Netflix since multi-region deployments of infra are quite common and region failover is a common mitigation strategy. 
With that in mind, it's a matter of time when quite a few jobs would require autoscaling during region failoves.